lib/repository/repository.ts

```diff
@@ -259,6 +259,43 @@ export class Repository {
     )
   }
 
+  /**
+   * Use Date object to set the {@link Entity}'s time to live. If the {@link Entity}
+   * is not found, does nothing.
+   *
+   * @param id The ID of the {@link Entity} to set an expiration date for.
+   * @param expirationDate The time the data should expire.
+   */
+  async expireAt(id: string, expirationDate: Date): Promise<void>;
+
+  /**
+   * Use Date object to set the {@link Entity | Entities} in Redis with the given
+   * ids. If a particular {@link Entity} is not found, does nothing.
+   *
+   * @param ids The IDs of the {@link Entity | Entities} to set an expiration date for.
+   * @param expirationDate The time the data should expire.
+   */
+  async expireAt(ids: string[], expirationDate: Date): Promise<void>;
+  
+  async expireAt(idOrIds: string | string[], expirationDate: Date) {
+    const ids = typeof idOrIds === 'string' ? [idOrIds] : idOrIds;
+    const timeNow: number = Date.now();
+    if (Date.now() >= expirationDate.getTime()) {
+      throw new Error(
+        `${expirationDate.toString()} is invalid. Expiration date must be in the future.`
+      );
+    }
+    const ttlInSeconds: number = Math.round(
+      (expirationDate.getTime() - timeNow) / 1000
+    );
+    await Promise.all(
+      ids.map((id) => {
+        const key = this.makeKey(id);
+        return this.client.expire(key, ttlInSeconds);
```

Comment:
There is a native `EXPIREAT` command in Redis. I think it would be more efficient and tidier to add a `.expireAt` method to `Client` and then call it.

That said, this implementation looks great!

Reply:
@guyroyse Thanks for reviewing and leaving feedback!

 I just updated the branch/PR to utilize the Redis' `EXPIREAT` via the client's `.expireAt` method and simplified the handling logic in the repository.

Please let me know if you think it need any other updates!